DQM/CTPPS/plugins/CTPPSRandomDQMSource.cc

```diff
@@ -0,0 +1,201 @@
+}
+
+//--------------------------------------------------------------------------
+
+void CTPPSRandomDQMSource::dqmBeginRun(edm::Run const &run, edm::EventSetup const &) {
+  for (int stn = 0; stn < StationIDMAX; stn++) {
+    StationStatus[stn] = 0;
+    for (int rp = 0; rp < RPotsIDMAX; rp++)
+      RPstatus[stn][rp] = 0;
+  }
+
+  unsigned int rpSts = rpStatusWord << 1;
+  for (int stn = 0; stn < NStationMAX; stn++) {
+    int stns = 0;
+    for (int rp = 0; rp < NRPotsMAX; rp++) {
+      rpSts = (rpSts >> 1);
+      RPstatus[stn][rp] = rpSts & 1;
+      if (RPstatus[stn][rp] > 0)
+        stns = 1;
+    }
+    StationStatus[stn] = stns;
+  }
+
+  for (int index = 0; index < 2 * 3 * NRPotsMAX; index++)
+    RPindexValid[index] = 0;
+}
+
+//-------------------------------------------------------------------------------------
+
+void CTPPSRandomDQMSource::bookHistograms(DQMStore::IBooker &ibooker, edm::Run const &, edm::EventSetup const &) {
+  ibooker.cd();
+  ibooker.setCurrentFolder("CTPPS/TrackingPixel");
+
+  hBX = ibooker.book1D("events per BX", "ctpps_pixel;Event.BX", 4002, -1.5, 4000. + 0.5);
+
+  for (int arm = 0; arm < NArms; arm++) {
+    CTPPSDetId ID(CTPPSDetId::sdTrackingPixel, arm, 0);
+    string sd;
+    ID.armName(sd, CTPPSDetId::nPath);
+
+    ibooker.setCurrentFolder(sd);
+
+    for (int stn = 0; stn < NStationMAX; stn++) {
+      if (StationStatus[stn] == 0)
+        continue;
+      ID.setStation(stn);
+      string stnd;
+      CTPPSDetId(ID.stationId()).stationName(stnd, CTPPSDetId::nPath);
+
+      ibooker.setCurrentFolder(stnd);
+
+      for (int rp = RPn_first; rp < RPn_last; rp++) {  // only installed pixel pots
+        ID.setRP(rp);
+        string rpd, rpTitle;
+        CTPPSDetId(ID.rpId()).rpName(rpTitle, CTPPSDetId::nFull);
+        CTPPSDetId(ID.rpId()).rpName(rpd, CTPPSDetId::nPath);
+
+        ibooker.setCurrentFolder(rpd);
+
+        int indexP = getRPindex(arm, stn, rp);
+        RPindexValid[indexP] = 1;
+
+        h2HitsVsBXRandoms[indexP] = ibooker.book2D("Digi per plane per BX - random triggers",
+                                                   rpTitle + ";Event.BX;Plane",
+                                                   4002,
+                                                   -1.5,
+                                                   4000. + 0.5,
+                                                   NplaneMAX,
+                                                   0,
+                                                   NplaneMAX);
+
+      }  // end for(int rp=0; rp<NRPotsMAX;...
+    }    // end of for(int stn=0; stn<
+  }      // end of for(int arm=0; arm<2;...
+
+  return;
+}
+
+//-------------------------------------------------------------------------------
+
+void CTPPSRandomDQMSource::analyze(edm::Event const &event, edm::EventSetup const &eventSetup) {
+  Handle<DetSetVector<CTPPSPixelDigi>> pixDigi;
+  event.getByToken(tokenDigi, pixDigi);
+
+  cout << "Before checking digi" << endl;
```

Comment:
`cout`s are not a good practise (and here they are even unguarded..). Use the standard CMSSW logging, e.g. `LogDebug` calls, see for example
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger

Reply:
Woops, my bad, that's not even useful, it was just for quick debugging. Removed!